code_share/app_code_share/urls.py

```diff
@@ -2,7 +2,7 @@
 from .views import *
 
 urlpatterns = [
-    url(r'^$', home, name='home'),
-    url(r'^(?P<hash_id>\d+)/$', view_by_hash, name='view_by_hash'),
-    url(r'^(?P<file_name>[\w-]+)/$', view_by_file, name='view_by_file')
+    url(r'^$', home, name='app_home'),
+    url(r'^app/(?P<hash_id>\d+)/$', view_by_hash, name='view_by_hash'),
```

Comment:
when i have a domain name like domain/filename    and i use domain/api its assume api as a filename and give an 404 so i made the url changes